mods/ctf/ctf_report/init.lua

```diff
@@ -0,0 +1,143 @@
+	local mods = get_irc_mods()
+	local idx = table.indexof(mods, name)
+	if idx > 0 then
+		table.remove(mods, idx)
+		storage:set_string("irc_mods", table.concat(mods, ","))
+		return true
+	end
+	return false
+end
+
+-- If word corresponds to a player name, this method returns the
+-- word enclosed by color escape codes of the player's team and
+-- also the name followed by team name in parenthesis for IRC
+local function parse_word(word)
+	if not minetest.get_player_by_name(word) then
+		return minetest.colorize("#FFFF00", word), word
+	end
+
+	local tplayer = ctf.player(word)
+	if not tplayer then
+		return minetest.colorize("#FFFF00", word), word
+	end
+
+	return minetest.colorize(ctf_colors.get_color(tplayer).css, word),
+			word .. " (team " .. (tplayer.team or "none") .. ")"
+end
+
+function ctf_report.send_report(report)
+	local ingame_report, irc_report
+
+	-- Check if player name(s) is mentioned.
+	-- colorize name for ingame report, and add
+	-- team name in parenthesis for irc report
+	local parts = report:split(" ")
+	for _, part in pairs(parts) do
+		local ingame_part, irc_part = parse_word(part)
+
+		if not ingame_report then
+			ingame_report = ingame_part
+		else
+			ingame_report = ingame_report .. " " .. ingame_part
+		end
+
+		if not irc_report then
+			irc_report = irc_part
+		else
+			irc_report = irc_report .. " " .. irc_part
+		end
+	end
+
+	-- Send to online moderators / admins
+	-- Get comma separated list of online moderators and admins
+	local mods = {}
+
+	for _, player in pairs(minetest.get_connected_players()) do
+		local toname = player:get_player_name()
+		if minetest.check_player_privs(toname, {kick = true, ban = true}) then
+			table.insert(mods, toname)
+			if ingame_report then
+				minetest.chat_send_player(toname, ingame_report)
+			end
+		end
+	end
+
+	if not minetest.global_exists("irc") then
+		return
+	end
+
+	-- Append list of moderators in-game
+	local msg
+	if #mods == 0 then
+		msg = irc_report .. " (no mods online)"
+	else
+		msg = irc_report .. " (mods online: " .. table.concat(mods, ", ") .. ")"
+	end
+
+	-- Send to IRC moderators
+	for _, toname in pairs(get_irc_mods()) do
+		if not minetest.get_player_by_name(toname) then
+			minetest.chat_send_player(toname, msg)
+		end
+	end
+end
+
+minetest.register_chatcommand("report", {
+	func = function(name, param)
+		param = param:trim()
+		if param == "" then
+			return false, "Please add a message to your report. " ..
+				"If it's about (a) particular player(s), please also include their name(s)."
+		end
+
+		local _, count = string.gsub(param, " ", "")
+		if count == 0 then
+			return false, "If you're reporting a player, you should" ..
+				" also include a reason why. (Eg: swearing, sabotage)"
+		end
+
+		param = name .. " reported: " .. param
+
+		ctf_report.send_report(param)
+
+		return true, "Reported. We'll get back to you."
+	end
+})
+
+minetest.register_chatcommand("report_sub", {
+	privs = { kick = true },
```

Comment:
IMO, spaces here not needed.
```suggestion
	privs = {kick = true},
```

Reply:
I'd prefer to retain these spaces, as they make the code more readable, IMO.

Reply:
Very well! 👍
They don't make anything different in-game (and it doesn't results in crashes), so, it works both ways.